samples/cpp/benchmark_app/main.cpp

```diff
@@ -378,11 +378,19 @@ int main(int argc, char* argv[]) {
         for (auto& device : devices) {
             auto& device_config = config[device];
 
+            ov::hint::PerformanceMode ov_perf_hint = ov::hint::PerformanceMode::THROUGHPUT;
+
             // high-level performance modes
             if (!device_config.count(ov::hint::performance_mode.name())) {
-                device_config.emplace(ov::hint::performance_mode(get_performance_hint(device, core)));
```

Comment:
There are usages of `ov::hint::PerformanceMode::UNDEFINED` left. Will you remove them in this PR?

Reply:
Since ov::hint::PerformanceMode::UNDEFINED will be kept until 2024, I will not remove it in this PR.

This PR is a follow-up to PR 16563, to avoid using ov::hint::PerformanceMode::UNDEFINED.